File: 0006-zigzag-conversion/0006-zigzag-conversion.js
var convert = function(s, numRows) {
    const n = s.length
    if(numRows == 1){
        return s
    }
    const res = new Array(numRows).fill("")
    let up = false
    let k = 1
    for(let i=0; i<n; i++){
        res[k-1] += s[i]
        if(k === numRows){
            up = true
        } else if(k === 1){
            up = false
        }
        
        k = up ? k-1 : k+1
    }
    return res.join("")
};

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if(numRows == 1):
            return s
        n = len(s)
        k = 1
        res = [""]*numRows
        down = True
        for i in range(n):
            res[k-1] += s[i]
            if(k == numRows):
                down = False
            elif(k == 1):
                down = True
                
            if(down):
                k+=1
            else:
                k-=1
        return "".join(res)